packages/date-picker/src/DatePicker/SingleDateContext/SingleDateContext.tsx

```diff
@@ -0,0 +1,187 @@
+import React, {
+  createContext,
+  PropsWithChildren,
+  SyntheticEvent,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import { usePrevious } from '@leafygreen-ui/hooks';
+
+import {
+  DateType,
+  getFirstOfMonth,
+  setToUTCMidnight,
+  useDatePickerContext,
+} from '../../shared';
+import { getISODate, isSameUTCDay } from '../../shared/utils';
+import { getInitialHighlight } from '../utils/getInitialHighlight';
+
+import {
+  SingleDateContextProps,
+  SingleDateProviderProps,
+} from './SingleDateContext.types';
+import { useDateRangeComponentRefs } from './useDatePickerComponentRefs';
+
+export const SingleDateContext = createContext<SingleDateContextProps>(
+  {} as SingleDateContextProps,
+);
+
+/**
+ * A provider for context values in a single DatePicker
+ */
+export const SingleDateProvider = ({
+  children,
+  value,
+  setValue: _setValue,
+  handleValidation,
+}: PropsWithChildren<SingleDateProviderProps>) => {
+  const refs = useDateRangeComponentRefs();
+  const { isOpen, setOpen } = useDatePickerContext();
+  const prevValue = usePrevious(value);
+
+  const today = useMemo(() => setToUTCMidnight(new Date(Date.now())), []);
```

Comment:
we might not want to memoize "today" as one interaction with the datepicker could span a few minutes (days - if 11:59 -> 12:02), and this could easily get stale.

Reply:
Shouldn't be too expensive. I'd want to add tests for this edge case though

Reply:
Ticket: [Remove memoization of today](https://jira.mongodb.org/browse/LG-3812)